From 8025f93fe7f0cb66a1f3d029a73067b095293e7f Mon Sep 17 00:00:00 2001
From: Andrii Sultanov <andriy.sultanov@vates.tech>
Date: Tue, 27 May 2025 14:38:39 +0100
Subject: [PATCH] Add qcow2 as supported format by xcp-rrdd-iostat

Since we are now supporting qcow file `tap-ctl list` can return strings like:
  - "1564848    0    0      qcow2 /var/run/sr-mount/..."
Without this patch the type "qcow2" is unknown and xcp-rrdd-iostat generates an
error like: returned a line that could not be parsed. Ignoring
This patch fixes the issue.

Signed-off-by: Guillaume <guillaume.thouvenin@vates.tech>
---
 ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml b/ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml
index 0f5470153..6141090ea 100644
--- a/ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml
+++ b/ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml
@@ -332,7 +332,10 @@ let refresh_phypath_to_sr_vdi () =
 let exec_tap_ctl_list () : ((string * string) * int) list =
   let tap_ctl = "/usr/sbin/tap-ctl list" in
   let extract_vdis pid minor _state kind phypath =
-    if not (kind = "vhd" || kind = "aio") then raise (Failure "Unknown type") ;
+    if not (kind = "vhd" || kind = "aio" || kind = "qcow2") then (
+      D.warn {|"%s" is not a known type.|} kind ;
+      raise (Failure "Unknown type")
+    ) ;
     (* Look up SR and VDI uuids from the physical path *)
     if not (Hashtbl.mem phypath_to_sr_vdi phypath) then
       refresh_phypath_to_sr_vdi () ;
